### ISSUE TITLE
BE03: Validate data cell appropriate with the title column

### ISSUE THREAD

**Issue by nnh53** [PiedTeam/DreameCloneTraining-BE-Project#13]
(US-03)

- Description:
  - Check the format of the cell compare to the title 
    - exp: title: "loại", data: "áo" -> correct 
    - exp: title: "loại", data: "13$" -> wrong
- Reason for the estimated point:
  - Have experience
- Approval Standards:
  - All the cell have to correct format

**Comment by nnh53** [comment]
![Image](https://github.com/PiedTeam/DreameCloneTraining-BE-Project/assets/113448161/b3270781-b292-409e-9007-a453ed722d7e)

- After this function we need 3 list: addNewRowList,  updateRowList,  ErrorRowList

**Comment by nnh53** [comment]
![Image](https://github.com/PiedTeam/DreameCloneTraining-BE-Project/assets/113448161/db5b1d99-afd8-419c-8518-3cdab1f481b7)

**Comment by nnh53** [comment]
![Image](https://github.com/PiedTeam/DreameCloneTraining-BE-Project/assets/113448161/61e1cb5f-883e-4540-83d6-503018fc83a8)

**Comment by lcaohoanq** [comment]
> ![Image](https://private-user-images.githubusercontent.com/113448161/320045427-b3270781-b292-409e-9007-a453ed722d7e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMxMDA0MTAsIm5iZiI6MTcxMzEwMDExMCwicGF0aCI6Ii8xMTM0NDgxNjEvMzIwMDQ1NDI3LWIzMjcwNzgxLWIyOTItNDA5ZS05MDA3LWE0NTNlZDcyMmQ3ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxNFQxMzA4MzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZjIxMzA5M2ZlNzMyYTYxMTRkMjgzMDkyOWVmNDA3MzQ0ZjMyMzkwMTNlODAwNDg5MWNhZWE4NTA3ZDliOGY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.usC7X9CXEg06My_Do0VtBt4B_0JYvauoeqLgQwfo6u8)
> 
> * Frontend need 3 list after this function: addNewRowList,  updateRowList,  ErrorRowList

@nnh53 
I don't really understand what you mean, does this my require as well or you just notify for FE team

**Comment by lcaohoanq** [comment]
@nnh53 
#6 not done yet, can i do on my own generated fake data?
- Employees Dreame.vi.xlsx
![image](https://github.com/PiedTeam/DreameCloneTraining-BE-Project/assets/136492579/04aaf1bb-aef3-46fb-8705-9943d77f9f77)
- Products Dreame.en.xlsx
![image](https://github.com/PiedTeam/DreameCloneTraining-BE-Project/assets/136492579/4ef36897-d7ee-4000-a9cc-0e39bbd8893a)

**Comment by nnh53** [comment]
@lcaohoanq do what ever u can to the task, urge @huy1707developer to do #6 faster

**Comment by lcaohoanq** [comment]
@nnh53 
Does this issues can be divide to smaller, the validate requirement is huge large, i don't know how to start with
- As my demo data for the `Products Dreame.en.xlsx`
![image](https://github.com/PiedTeam/DreameCloneTraining-BE-Project/assets/136492579/09d48cfc-3fb6-441c-a4ce-a49fe4133f27)
- The data type of data when read file can be a string, number base on `xlsx` default, does i need to check the data type as well, the requirement do not specify this (i'm do not have the legit data, now just on demo and when i can get the real data?)
> Requirement
- exp: title: "loại", data: "áo" -> correct
   - Now with demo of `Product Dreame.en.xlsx` and  `Employees Dreame.vi.xlsx`, will have other file type also, i mean when loại -> áo, product (may be split in smaller like vaccum cleaner, robot,... ), nhân viên (admin, manager, dev, sales,...), i need to group all these data 
- exp: title: "tiền", data: "vnd" -> correct
   - Which is the data in Price column ($99.99 or 99,99 only), (100.000 VND or 100.000 VNĐ or 100.000 only) 
- exp: title: "price", data: "$" -> correct
   - The price column when see in excel it become: `$99.99` but when read file it only `99.99`, mean that the currency are handling by excel as well
- exp: title: "loại", data: "13$" -> wrong
   - As above, the legit data we still not have, demo for this is huge difficult

**Comment by nnh53** [comment]
@lcaohoanq yes, in the next breakdown metting we will discuss about this

**Comment by hoangday185** [comment]
For non-mandatory data fields, if the data field is empty, the empty string is returned, otherwise it is checked as usual

**Comment by nnh53** [comment]
- Break down to multiple smaller task

**Comment by hoangday185** [comment]
We have to manage objects like model and manufacturer. When we add a new object in the excel file, we will ask the user if they want to add a new one. If not, ignore that product. If so, popup and add it.

**Comment by lcaohoanq** [comment]
@nnh53 @hoangday185 @huy1707developer 
The title will be lower case only, instead of capitalize the first word


![Image](https://github.com/PiedTeam/DreameCloneTraining-BE-Project/assets/136492579/f5dd8f98-b11e-419e-916a-57d2a791b49b)

**Comment by nnh53** [comment]
No, let the user choice what ever they want, we will lower in BE code

**Comment by lcaohoanq** [comment]
> No, let the user choice however they want, we will lower in BE code @lcaohoanq

Yes, base on user choice, i just want to notify everyone that i will work on the sample data fix to lower case the title
